airflow/models/xcom.py

```diff
@@ -223,7 +223,9 @@ def serialize_value(value: Any):
         if conf.getboolean('core', 'enable_xcom_pickling'):
             return pickle.dumps(value)
         try:
-            return json.dumps(value).encode('UTF-8')
+            from airflow.serialization.serialized_objects import BaseSerialization
+
+            return json.dumps(BaseSerialization._serialize(value, fail_on_error=True)).encode('UTF-8')
```

Comment:
Maybe we should add a `BaseSerialization.to_json()` method for symmetry

Reply:
It already exists.

Reply:
Oh, excetp:

```python
    @classmethod
    def to_json(cls, var: Union[DAG, BaseOperator, dict, list, set, tuple]) -> str:
        """Stringifies DAGs and operators contained by var and returns a JSON string of var."""
        return json.dumps(cls.to_dict(var), ensure_ascii=True)

    @classmethod
    def to_dict(cls, var: Union[DAG, BaseOperator, dict, list, set, tuple]) -> dict:
        """Stringifies DAGs and operators contained by var and returns a dict of var."""
        # Don't call on this class directly - only SerializedDAG or
        # SerializedBaseOperator should be used as the "entrypoint"
        raise NotImplementedError()

```